httpclient/config.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"github.com/acronis/go-appkit/config"
+	"time"
+)
+
+const (
+	// DefaultLimit is the default limit for rate limiting.
+	DefaultLimit = 10
+
+	// DefaultBurst is the default burst for rate limiting.
+	DefaultBurst = 100
+
+	// DefaultWaitTimeout is the default wait timeout for rate limiting.
+	DefaultWaitTimeout = 10 * time.Second
+
+	// configuration properties
+	cfgKeyRetriesMax            = "retries.max_retry_attempts"
+	cfgKeyRateLimitsLimit       = "rate_limits.limit"
+	cfgKeyRateLimitsBurst       = "rate_limits.burst"
+	cfgKeyRateLimitsWaitTimeout = "rate_limits.wait_timeout"
+	cfgKeyTimeout               = "timeout"
+)
+
+// RateLimitConfig represents configuration options for HTTP client rate limits.
+type RateLimitConfig struct {
+	Limit       int           `mapstructure:"limit"`
+	Burst       int           `mapstructure:"burst"`
+	WaitTimeout time.Duration `mapstructure:"wait_timeout"`
+}
+
+// Set is part of config interface implementation.
+func (c *RateLimitConfig) Set(dp config.DataProvider) (err error) {
+	limit, err := dp.GetInt(cfgKeyRateLimitsLimit)
+	if err != nil {
+		return err
+	}
+	c.Limit = limit
```

Comment:
Let's do more validation and return an error for example if the limit is negative.